### PR TITLE
Whack-a-mole 7: The last whack-a-mole?

### PR DESCRIPTION
**Pull request by Manishearth**
We're almost done.

This contains some scriptextensions changes, but I think we need to go back and fix up icu::properties::Script first and then make an FFI that matches it. Happy to remove the ignores for that.

### CODE REVIEW

File: ffi/diplomat/ffi_coverage/src/main.rs [hunk 1]
```diff
@@ -253,6 +257,14 @@ lazy_static::lazy_static! {
...
         "icu::properties::WordBreak",
 
+        // ScriptWithExtensions is tricky due to the borrow and DataPayload
+        "icu::properties::script::ScriptExtensionsSet",
+        "icu::properties::script::ScriptWithExtensions::get_script_extensions_set",
+        "icu::properties::script::ScriptWithExtensions::get_script_extensions_val",
+
+        // ranges api
+        "icu::properties::script::ScriptWithExtensions::get_script_extensions_ranges",
+
```
Comment (sffc) [inline ffi/diplomat/ffi_coverage/src/main.rs:267]: Please don't add these suppressions

File: ffi/diplomat/src/script.rs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    use diplomat_runtime::DiplomatResult;
+
+    #[diplomat::opaque]
+    /// An ICU4X ScriptWithExtensions map object, capable of holding a map of codepoints to scriptextensions values
+    #[diplomat::rust_link(icu::properties::script::ScriptWithExtensions, Struct)]
+    pub struct ICU4XScriptWithExtensionsSet(
+        pub DataPayload<provider::ScriptWithExtensionsPropertyV1Marker>,
+    );
+
+    impl ICU4XScriptWithExtensionsSet {
```
Comment (sffc) [inline ffi/diplomat/src/script.rs:22]: Nit: Misnomer. Just call it `ICU4XScriptWithExtensions`. There's a reason SWE and SES are different in Rust

File: ffi/diplomat/ffi_coverage/src/main.rs [hunk 3]
```diff
@@ -259,13 +263,15 @@ lazy_static::lazy_static! {
...
         "icu::properties::sets::CodePointSetDataBorrowed::iter_ranges",
+        "icu::properties::script::ScriptWithExtensions::get_script_extensions_ranges",
         "icu::properties::maps::CodePointMapData::as_code_point_trie",
         "icu::properties::maps::CodePointMapData::from_code_point_trie",
         "icu::properties::sets::CodePointSetData::as_code_point_inversion_list",
         "icu::properties::sets::CodePointSetData::from_code_point_inversion_list",
         "icu::properties::sets::CodePointSetData::to_code_point_inversion_list",
         "icu::collections::codepointinvlist",
         "icu::collections::codepointtrie",
+        "icu::collections::char16trie",
```
Comment (sffc) [inline ffi/diplomat/ffi_coverage/src/main.rs:274]: Question: When did we decide to exclude the three collection types? Seems like those are useful over FFI. Is there something that makes them tricky?

File: ffi/diplomat/ffi_coverage/src/main.rs [hunk 4]
```diff
@@ -241,6 +241,10 @@ lazy_static::lazy_static! {
         // could be exposed in the future but it's complicated
         "icu::plurals::rules",
 
+        // Pulls in libstd, which we'd rather not do, and it's not a super prominent
+        // API
```
Comment (sffc) [inline ffi/diplomat/ffi_coverage/src/main.rs:245]: Nit: Don't editorialize on the "prominence" of the APIs

```suggestion
        // Pulls in libstd, which we'd rather not do
```

**Review by sffc — APPROVED** [review]

File: ffi/diplomat/ffi_coverage/src/main.rs (hunk 3 above)
Comment (Manishearth) [reply ffi/diplomat/ffi_coverage/src/main.rs:273]: I think the decision was to support building these, but I'm not sure if that's for 1.0, and I think the FFI for that is rather tricky. We can un-ignorelist, I would just rather we exposed this after 1.0 in a nicer way.

**Review by sffc — APPROVED** [review]

**Pull request merged**